src/v/config/client_group_byte_rate_quota.h

```diff
@@ -0,0 +1,73 @@
+/*
+ * Copyright 2022 Redpanda Data, Inc.
+ *
+ * Use of this software is governed by the Business Source License
+ * included in the file licenses/BSL.md
+ *
+ * As of the Change Date specified in that file, in accordance with
+ * the Business Source License, use of this software will be governed
+ * by the Apache License, Version 2.0
+ */
+
+#pragma once
+
+#include "config/convert.h"
+#include "config/property.h"
+#include "json/_include_first.h"
+#include "json/stringbuffer.h"
+#include "json/writer.h"
+
+#include <seastar/core/sstring.hh>
+
+#include <yaml-cpp/node/node.h>
+
+#include <iosfwd>
+#include <string>
+
+namespace config {
+
+struct client_group_quota {
+    using key_type = ss::sstring;
+
+    ss::sstring group_name;
+    ss::sstring clients_prefix;
+    int64_t quota;
+
+    static ss::sstring key_name() { return "group_name"; }
+
+    ss::sstring key() const { return group_name; }
+
+    friend bool operator==(const client_group_quota&, const client_group_quota&)
+      = default;
+
+    friend std::ostream&
+    operator<<(std::ostream& os, const client_group_quota& ep);
+};
+
+namespace detail {
+
+template<>
+consteval std::string_view property_type_name<client_group_quota>() {
+    return "config::client_group_quota";
+}
+
+} // namespace detail
+
+} // namespace config
+
+namespace YAML {
```

Comment:
I feel like this commit can be split into two. One that determines the new validator and another that defines the YAML over loads to convert from YAML into the well defined type

Reply:
I think that they can be in same commit.
This Yaml is needed only in certain configuration right now. To fully understand commit I want to have all converts and configuration in same diff